Traceback (most recent call last):
  File "/home/hoffmann/.local/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/home/hoffmann/.local/lib/python3.8/site-packages/nbclient/client.py", line 1269, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/home/hoffmann/.local/lib/python3.8/site-packages/nbclient/util.py", line 85, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/home/hoffmann/.local/lib/python3.8/site-packages/nbclient/util.py", line 60, in just_run
    return loop.run_until_complete(coro)
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/hoffmann/.local/lib/python3.8/site-packages/nbclient/client.py", line 707, in async_execute
    await self.async_execute_cell(
  File "/home/hoffmann/.local/lib/python3.8/site-packages/nbclient/client.py", line 1025, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/home/hoffmann/.local/lib/python3.8/site-packages/nbclient/client.py", line 919, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
#import stuff
import numpy as np
import matplotlib.pyplot as plt

#assigning the number of clusters
K = 2

#assigning the number of cluster centroids
centroids = np.random.rand(K,2)

#plot the data points and the centroids
plt.scatter(X[:,0],X[:,1])
plt.scatter(centroids[:,0],centroids[:,1],c='r',s=100)
plt.show()

------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mNameError[0m                                 Traceback (most recent call last)
[0;32m<ipython-input-1-93f5bca78ca5>[0m in [0;36m<module>[0;34m[0m
[1;32m     10[0m [0;34m[0m[0m
[1;32m     11[0m [0;31m#plot the data points and the centroids[0m[0;34m[0m[0;34m[0m[0;34m[0m[0m
[0;32m---> 12[0;31m [0mplt[0m[0;34m.[0m[0mscatter[0m[0;34m([0m[0mX[0m[0;34m[[0m[0;34m:[0m[0;34m,[0m[0;36m0[0m[0;34m][0m[0;34m,[0m[0mX[0m[0;34m[[0m[0;34m:[0m[0;34m,[0m[0;36m1[0m[0;34m][0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     13[0m [0mplt[0m[0;34m.[0m[0mscatter[0m[0;34m([0m[0mcentroids[0m[0;34m[[0m[0;34m:[0m[0;34m,[0m[0;36m0[0m[0;34m][0m[0;34m,[0m[0mcentroids[0m[0;34m[[0m[0;34m:[0m[0;34m,[0m[0;36m1[0m[0;34m][0m[0;34m,[0m[0mc[0m[0;34m=[0m[0;34m'r'[0m[0;34m,[0m[0ms[0m[0;34m=[0m[0;36m100[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m     14[0m [0mplt[0m[0;34m.[0m[0mshow[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;31mNameError[0m: name 'X' is not defined
NameError: name 'X' is not defined

